e2e/interchain_accounts_test.go

```diff
@@ -0,0 +1,220 @@
+		})
+	})
+}
+
+func (s *InterchainAccountsTestSuite) TestMsgSubmitTx_FailedTransfer_InsufficientFunds() {
+	t := s.T()
+	ctx := context.TODO()
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	connectionId := "connection-0"
+
+	// setup 2 accounts: controller account on chain A, a second chain B account.
+	// host account will be created when the ICA is registered
+	controllerAccount := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainBAccount := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	var hostAccount string
+
+	t.Run("register interchain account", func(t *testing.T) {
+		err := s.RegisterICA(ctx, chainA, controllerAccount, controllerAccount.Bech32Address(chainA.Config().Bech32Prefix), connectionId)
+		s.Require().NoError(err)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("verify interchain account", func(t *testing.T) {
+		var err error
+		hostAccount, err = s.QueryInterchainAccount(ctx, chainA, controllerAccount.Bech32Address(chainA.Config().Bech32Prefix), connectionId)
+		s.Require().NoError(err)
+		s.Require().NotZero(len(hostAccount))
+
+		channels, err := relayer.GetChannels(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID)
+		s.Require().NoError(err)
+		s.Require().Equal(len(channels), 2)
+	})
+
+	t.Run("fail to execute bank transfer over ICA", func(t *testing.T) {
+		t.Run("verify empty host wallet", func(t *testing.T) {
+			hostAccountBalance, err := chainB.GetBalance(ctx, hostAccount, chainB.Config().Denom)
+			s.Require().NoError(err)
+			s.Require().Zero(hostAccountBalance)
+		})
+
+		t.Run("broadcast MsgSubmitTx", func(t *testing.T) {
+			// assemble bank transfer message from host account to user account on host chain
+			transferMsg := &banktypes.MsgSend{
+				FromAddress: hostAccount,
+				ToAddress:   chainBAccount.Bech32Address(chainB.Config().Bech32Prefix),
+				Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+			}
+
+			// assemble submitMessage tx for intertx
+			submitMsg, err := intertxtypes.NewMsgSubmitTx(
+				transferMsg,
+				connectionId,
+				controllerAccount.Bech32Address(chainA.Config().Bech32Prefix),
+			)
+			s.Require().NoError(err)
+
+			// broadcast submitMessage tx from controller account on chain A
+			// this message should trigger the sending of an ICA packet over channel-1 (channel created between controller and host)
+			// this ICA packet contains the assembled bank transfer message from above, which will be executed by the host account on the host chain.
+			resp, err := s.BroadcastMessages(
+				ctx,
+				chainA,
+				controllerAccount,
+				submitMsg,
+			)
+
+			s.AssertValidTxResponse(resp)
+			s.Require().NoError(err)
+		})
+
+		t.Run("verify balance is the same", func(t *testing.T) {
```

Comment:
The test above adds a wait for 10 blocks. Maybe we should add an assertion that the packet was relayed?

Reply:
Great suggestion, will add this.